### PR TITLE
Add support for coordinates resolution.

### PR DESCRIPTION
**Pull request by keyurva**
* For #203 
* Coordinates resolution is only enabled if a new `--coordinates-resolution` flag is set.
* The implementation is intentionally minimally invasive to the existing resolution code path and is executed only when the above flag is set.
* Currently it only supports numeric `latitude` and `longitude` properties and basic validation. Support for more formats and validations will be added subsequently.

### CODE REVIEW

File: util/src/main/java/org/datacommons/util/ReconClient.java [hunk 1]
```diff
@@ -1,51 +1,130 @@
...
- * <p>Currently it only resolves coordinates.
+ * <p>Currently it only resolves coordinates. If the number of coordinates to resolve are greater
+ * than {@code chunkSize}, the API calls will be partitioned into max {@code chunkSize}d batches.
  */
 public class ReconClient {
   private static final String RESOLVE_COORDINATES_API_URL =
       "https://api.datacommons.org/v1/recon/resolve/coordinate";
 
+  private static final String RESOLVE_ENTITIES_API_URL =
+      "https://api.datacommons.org/v1/recon/entity/resolve";
```
Comment (pradh) [inline util/src/main/java/org/datacommons/util/ReconClient.java:33]: Is this used?

File: util/src/main/java/org/datacommons/util/ReconClient.java [hunk 2]
```diff
@@ -1,51 +1,130 @@
...
+        .thenApply(
+            chunkedResponses ->
+                ResolveCoordinatesResponse.newBuilder()
+                    .addAllPlaceCoordinates(
+                        chunkedResponses.stream()
+                            .flatMap(
+                                chunkedResponse ->
+                                    chunkedResponse.getPlaceCoordinatesList().stream())
+                            .collect(toList()))
+                    .build());
```
Comment (pradh) [inline util/src/main/java/org/datacommons/util/ReconClient.java:88]: please add some comments for this block?

File: util/src/main/java/org/datacommons/util/CoordinatesResolver.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import org.datacommons.proto.Mcf.McfGraph.PropertyValues;
+import org.datacommons.proto.Mcf.McfGraph.Values;
+import org.datacommons.proto.Mcf.ValueType;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest;
+import org.datacommons.proto.Recon.ResolveCoordinatesRequest.Coordinate;
+import org.datacommons.proto.Recon.ResolveCoordinatesResponse;
+import org.datacommons.proto.Recon.ResolveCoordinatesResponse.Place;
+
+/** Resolves nodes with lat-lngs by calling the DC coordinates resolution API. */
```
Comment (pradh) [inline util/src/main/java/org/datacommons/util/CoordinatesResolver.java:15]: Add TODO to update counters on errors?

File: util/src/main/java/org/datacommons/util/CoordinatesResolver.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      new ConcurrentHashMap<>();
+
+  private final ReconClient client;
+
+  CoordinatesResolver(ReconClient client) {
+    this.client = client;
+  }
+
+  boolean submit(PropertyValues node) {
+    return getCoordinate(node).map(resolveCoordinates::add).map(unused -> true).orElse(false);
```
Comment (pradh) [inline util/src/main/java/org/datacommons/util/CoordinatesResolver.java:29]: nit: This looks like a slightly convoluted syntax to add to a set?

**Review by pradh — APPROVED** [review]
Nice!

Please consider updating docs in a follow up.

https://github.com/datacommonsorg/import/tree/master/docs

File: util/src/main/java/org/datacommons/util/ReconClient.java (hunk 1 above)
Comment (keyurva) [reply util/src/main/java/org/datacommons/util/ReconClient.java]: Good catch. Removed.

File: util/src/main/java/org/datacommons/util/CoordinatesResolver.java (hunk 3 above)
Comment (keyurva) [reply util/src/main/java/org/datacommons/util/CoordinatesResolver.java:15]: Done.

File: util/src/main/java/org/datacommons/util/CoordinatesResolver.java (hunk 4 above)
Comment (keyurva) [reply util/src/main/java/org/datacommons/util/CoordinatesResolver.java]: Updated. I typically don't like to use get() on Optional but it makes sense to do here for readability. Thanks for calling it out.

**Pull request merged**